app/src/components/avatar.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{
+    components::avatar_event_bus::AvatarEventBus,
+    infra::common_component::{CommonComponent, CommonComponentParts}};
+use anyhow::{bail, Result};
+use graphql_client::GraphQLQuery;
+use yew::{prelude::*};
+use yew_agent::{Bridge, Bridged};
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "queries/get_user_avatar.graphql",
+    response_derives = "Debug",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+pub struct GetUserAvatar;
+
+pub struct Avatar {
+    common: CommonComponentParts<Self>,
+    /// The user info. If none, the error is in `error`. If `error` is None, then we haven't
+    /// received the server response yet.
+    avatar: Option<String>,    
+    _producer: Box<dyn Bridge<AvatarEventBus>>,
+}
+
+/// State machine describing the possible transitions of the component state.
+/// It starts out by fetching the user's details from the backend when loading.
+pub enum Msg {
+    /// Received the user details response, either the user data or an error.
+    UserAvatarResponse(Result<get_user_avatar::ResponseData>),
+    Update
+}
+
+#[derive(yew::Properties, Clone, PartialEq, Eq)]
+pub struct Props {
+    pub username: String,
+    pub width: i32,
+    pub height: i32,
+}
+
+impl CommonComponent<Avatar> for Avatar {
+    fn handle_msg(&mut self, ctx: &Context<Self>, msg: <Self as Component>::Message) -> Result<bool> {
+        match msg {
+            Msg::UserAvatarResponse(response) => match response {
+                Ok(user) => self.avatar = user.user.avatar,
+                Err(e) => {
+                    self.avatar = None;
+                    bail!("Error getting user avatar: {}", e);
+                }
+            },
+            Msg::Update => self.get_user_avatar(ctx),
+        }
+        Ok(true)
```

Comment:
Technically this will cause an unnecessary redraw even if the avatar was None (not received yet) and we got None (user doesn't have an avatar).

Reply:
I'll see about adding logic to check if the avatar has changed or not

Reply:
So i want to be cautious of futuure work for allowing avatars to be deleted. I did some smrter decisions around redrawing though